### ISSUE TITLE
unescaped regular expressions

### ISSUE THREAD

**Issue by BenLubar** [julianlam/nodebb-plugin-mentions#83]
https://what.thedailywtf.com/topic/20703/template-js-testing

I can use `@ben..........lubar` to mention a user named `ben lubar`, but I can also type `@ben "HI MOM" lubar` in the same post and it'll be converted to a mention.

**Comment by julianlam** [comment]
Yes, that is potentially an issue... we take the match, and slugify it, so `ben.........lubar` is collapsed to `ben-lubar` which is a valid slug. 

:thinking: Opinions welcome.
